fastapi/encoders.py

```diff
@@ -95,7 +95,7 @@ def jsonable_encoder(
                 )
                 encoded_dict[encoded_key] = encoded_value
         return encoded_dict
-    if isinstance(obj, (list, set, frozenset, GeneratorType, tuple)):
+    if isinstance(obj, (list, set, frozenset, GeneratorType, tuple, Iterator)):
```

Comment:
More-proper question: Why don't we just make this `isinstance(obj, Iterable)`?

```python
from typing import Iterable

def gen(arr):
    for x in arr:
        yield x

l = [1, 2, 3, 4]
g = gen(l)
s = set(l)
t = tuple(l)
f = frozenset(l)

print(isinstance(g, Iterable))
print(isinstance(l, Iterable))
print(isinstance(s, Iterable))
print(isinstance(t, Iterable))
print(isinstance(f, Iterable))

# Output is all True
```

Reply:
The problem is exactly the one you guessed - Iterable makes some tests fail (https://github.com/tiangolo/fastapi/blob/22528373bba6a654323de416ad5c867cbadb81bb/tests/test_jsonable_encoder.py#L102). On the other `Iterator` maintains the current interface and fix the issue you opened, and in general the behaviour with Iterable[] types.